tests/console/validate_packages_and_patterns.pm

```diff
@@ -35,35 +35,20 @@ if (check_var('VALIDATE_PCM_PATTERN', 'azure')) {
     };
 }
 elsif (check_var('VALIDATE_PCM_PATTERN', 'aws')) {
-    # We have different pattern names on SLE 15 and SLE 12
-    my $aws_pattern = is_sle('15+') ? 'Amazon_Web_Services' : 'Amazon-Web-Services';
-    my @aws_pattern_packages = qw(
-      aws-cli cloud-init
-      cloud-regionsrv-client
-      docker-img-store-setup growpart supportutils-plugin-suse-public-cloud
-      regionServiceClientConfigEC2 s3fs
-    );
-    if (is_sle('15+')) {
-        push @aws_pattern_packages, qw(
-          cloud-regionsrv-client-plugin-ec2
-          patterns-public-cloud-15-Amazon-Web-Services
-          python3-ec2deprecateimg python3-ec2metadata python3-ec2publishimg
-          python3-ec2uploadimg python3-s3transfer python3-susepubliccloudinfo
-        );
-    }
-    else {
-        push @aws_pattern_packages, qw(
-          patterns-public-cloud-Amazon-Web-Services
-          python-ec2deprecateimg python-ec2metadata python-ec2publishimg
-          python-ec2uploadimg python-s3transfer python-susepubliccloudinfo
-        );
-    }
-    $software{$aws_pattern} = {
+    # Different pattern and packages names on SLE 15 and SLE 12
+    my %aws_specific = is_sle('15+') ? (name => 'Amazon_Web_Services', sle_version => '15-', py_version => 'python3') :
+      (name => 'Amazon-Web-Services', py_version => 'python');
+    $software{%aws_specific{name}} = {
         repo      => 'Module-Public-Cloud',
         installed => 1,
         condition => sub { check_var_array('PATTERNS', 'aws') },
         pattern   => 1,
-        packages  => \@aws_pattern_packages,
+        packages  => ['aws-cli', 'cloud-init', 'cloud-regionsrv-client', 'cloud-regionsrv-client-plugin-ec2',
+            'docker-img-store-setup', 'growpart', 'patterns-public-cloud-' . $aws_specific{sle_version} . 'Amazon-Web-Services',
+            $aws_specific{py_version} . '-ec2deprecateimg', $aws_specific{py_version} . '-ec2metadata',
```

Comment:
I was also concerned how strict we should be with packages names, so that it's not pain to maintain. So, let's try it put and see if it worth it, or we can be less strict with search.

Reply:
I agree, it is a refactor to avoid so many lines of code and conditions, but if it is painful to maintain we can do it in different way in the future.